lib/gooddata/lcm/actions/update_metric_formats.rb

```diff
@@ -0,0 +1,159 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+# Copyright (c) 2010-2021 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require_relative 'base_action'
+
+module GoodData
+  module LCM2
+    class UpdateMetricFormats < BaseAction
+      DESCRIPTION = 'Localize Metric Formats'
+
+      PARAMS = define_params(self) do
+        description 'Synchronization Info'
+        param :synchronize, array_of(instance_of(Type::SynchronizationInfoType)), required: true, generated: true
+
+        description 'Client Used for Connecting to GD'
+        param :gdc_gd_client, instance_of(Type::GdClientType), required: true
+
+        description 'Organization Name'
+        param :organization, instance_of(Type::StringType), required: false
+
+        description 'DataProduct to manage'
+        param :data_product, instance_of(Type::GDDataProductType), required: false
+
+        description 'Logger'
+        param :gdc_logger, instance_of(Type::GdLogger), required: true
+
+        description 'ADS Client'
+        param :ads_client, instance_of(Type::AdsClientType), required: false
+
+        description 'Input Source'
+        param :input_source, instance_of(Type::HashType), required: false
+
+        description 'Localization query'
+        param :localization_query, instance_of(Type::StringType), required: false
+      end
+
+      RESULT_HEADER = %i[client tag format]
+
+      class << self
+        def load_metric_data(params)
+          metric_input_source = validate_input_source(params.input_source)
+          if metric_input_source.nil?
+            GoodData.logger.info('No metric config or it is incorrect, skip metric updating')
+            return nil
+          end
+
+          metric_data_source = GoodData::Helpers::DataSource.new(metric_input_source)
+          begin
+            temp_csv = without_check(PARAMS, params) do
+              File.open(metric_data_source.realize(params), 'r:UTF-8')
+            end
+          rescue StandardError => e
+            GoodData.logger.warn("Unable to get metric data, skip updating: #{e.message}. Error: #{e}")
```

Comment:
Unable to get metric input source, skip updating metric formats. Error: ...

Reply:
Changed this log.